### DISCUSSION TITLE
SvgUri vs SvgFromUri

### DISCUSSION THREAD

**Discussion by pretorh** [software-mansion/react-native-svg#2574] [Q&A] [answered]
We have been using `<SvgUri uri={...} />` to load and display svg images from a site. However this is sometimes causing a captcha page to be loaded, which then causes an app crash.
This happens when `SvgUri` calls `SvgXml` with the "xml" (really the html error page), and that fails when `parse`ing the data.

Wrapping `parse` in a try catch and rendering a `null` fixed the crash.

But there are `SvgFromUri` and `SvgFromXml` classes in `src/xml.tsx`, and `SvgFromXml` already [handles parsing errors](https://github.com/software-mansion/react-native-svg/blob/main/src/xml.tsx#L124).

We are on an old version of `react-native-svg`, but similar functions and classes exist on `main`.

What is the difference between `SvgUri`/`SvgXml` and `SvgFromUri`/`SvgFromXml`?

**Comment by jakex7** [comment, marked as answer]
Actually, there isn't any difference, except that `*From*` is a class component, whereas the classic ones are functional components. This is done for compatibility reasons and because extending Component is required for Animated support.
